### PR TITLE
REF: move Resolution to tslibs.dtypes

### PR DESCRIPTION
**Pull request by jbrockmendel**
This is pretty much a clean move, in preparation for making FreqGroup an enum and trying to de-duplicate our 3+ enum-like classes

### CODE REVIEW

**Comment by jorisvandenbossche** [comment]
At some point in one of the previous PRs, you said that deduplication might not be possible after all (https://github.com/pandas-dev/pandas/pull/34587#issuecomment-639530365). Can you give a bit of context on what your idea / goal is on that now?

File: pandas/_libs/tslibs/dtypes.pyx [hunk 1]
```diff
@@ -127,6 +131,7 @@ _attrname_to_abbrevs = {
     "nanosecond": "N",
 }
 cdef dict attrname_to_abbrevs = _attrname_to_abbrevs
+cdef dict _abbrev_to_attrnames = {v: k for k, v in attrname_to_abbrevs.items()}
```
Comment (jreback) [inline pandas/_libs/tslibs/dtypes.pyx:134]: should make these both private or public

Comment (jreback) [reply pandas/_libs/tslibs/dtypes.pyx:134]: followon ok

**Review by jreback — APPROVED** [review]

File: pandas/_libs/tslibs/dtypes.pyx (hunk 1 above)
Comment (jbrockmendel) [reply pandas/_libs/tslibs/dtypes.pyx:134]: the public one is used outside this module, the private one is not

**Comment by jbrockmendel** [comment]
> Can you give a bit of context on what your idea / goal is on that now?

The mapping between FreqGroup and Resolution is not going to be a bijection, but im now thinking that both FreqGroup.FR_DAY and FreqGroup.FR_BUS map to Resolution.RESO_DAY.  This will allow us to clean up semi-kludge [here](https://github.com/pandas-dev/pandas/blob/master/pandas/core/indexes/period.py#L520).

**Pull request merged**